pkg/controller/kubemon/statefulset.go

```diff
@@ -0,0 +1,281 @@
+			FailureThreshold:    2,
+		},
+	}
+}
+func buildLabels(instance *dynatracev1alpha1.DynaKube) map[string]string {
+	return MergeLabels(instance.Labels,
+		BuildLabelsFromInstance(instance),
+		instance.Spec.KubernetesMonitoringSpec.Labels)
+}
+
+func buildLabelSelector(instance *dynatracev1alpha1.DynaKube) *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: MergeLabels(
+			BuildLabelsFromInstance(instance),
+			instance.Spec.KubernetesMonitoringSpec.Labels),
+	}
+}
+
+func buildKubernetesExpression(archKey string, osKey string) []corev1.NodeSelectorRequirement {
+	return []corev1.NodeSelectorRequirement{
+		{
+			Key:      archKey,
+			Operator: corev1.NodeSelectorOpIn,
+			Values:   []string{AMD64, ARM64},
+		},
+		{
+			Key:      osKey,
+			Operator: corev1.NodeSelectorOpIn,
+			Values:   []string{LINUX},
+		},
+	}
+}
+
+func buildVolumes(instance *dynatracev1alpha1.DynaKube) []corev1.Volume {
+	var volumes []corev1.Volume
+
+	customProperties := instance.Spec.KubernetesMonitoringSpec.CustomProperties
+	if !isCustomPropertiesNilOrEmpty(customProperties) {
+		valueFrom := instance.Spec.KubernetesMonitoringSpec.CustomProperties.ValueFrom
+		if valueFrom == "" {
+			valueFrom = fmt.Sprintf("%s-kubernetes-monitoring-%s", instance.Name, customproperties.Suffix)
+		}
+
+		volumes = append(volumes, corev1.Volume{
+			Name: customproperties.VolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: valueFrom,
+					Items: []corev1.KeyToPath{
+						{Key: customproperties.DataKey, Path: customproperties.DataPath},
+					},
+				},
+			},
+		})
+	}
+
+	return volumes
+}
+
+func isCustomPropertiesNilOrEmpty(customProperties *dynatracev1alpha1.DynaKubeValueSource) bool {
+	return customProperties == nil ||
+		(customProperties.Value == "" &&
+			customProperties.ValueFrom == "")
+}
+
+func buildEnvs(instance *dynatracev1alpha1.DynaKube, kubeSystemUID types.UID) []corev1.EnvVar {
+	envVars := []corev1.EnvVar{
+		{Name: DTCapabilities, Value: CapabilityEnv},
+		{Name: DTIdSeedNamespace, Value: instance.Namespace},
+		{Name: DTIdSeedClusterId, Value: string(kubeSystemUID)},
+	}
+
+	envVars = append(envVars, instance.Spec.KubernetesMonitoringSpec.Env...)
+
+	proxy := instance.Spec.Proxy
+	if !isProxyNilOrEmpty(proxy) {
+		var proxyEnvVar corev1.EnvVar
+
+		if proxy.ValueFrom != "" {
+			proxyEnvVar = corev1.EnvVar{
+				Name: ProxyEnv,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{Name: proxy.ValueFrom},
+						Key:                  ProxyKey,
+					},
+				},
+			}
+		} else {
+			proxyEnvVar = corev1.EnvVar{
+				Name:  ProxyEnv,
+				Value: proxy.Value,
+			}
+		}
+
+		envVars = append(envVars, proxyEnvVar)
+	}
+
+	return envVars
+}
+
+func buildArgs(instance *dynatracev1alpha1.DynaKube) []string {
+	args := []string{
+		DTTenantArg,
+		DTTokenArg,
+		DTServerArg,
```

Comment:
These aren't needed anymore (and the env vars are gone.)

Reply:
I added Marco's changes which removed the tenant specific env vars.
removed the arguments as well